src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -86,24 +104,85 @@ def add_helm_repo(kube_config, kube_context):
+        raise CLIError("No content was found in helm registry path response.")
+
+
+def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
+    if isinstance(ex, AuthenticationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Authentication error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, TokenExpiredError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Token expiration error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, HttpOperationError):
+        status_code = ex.response.status_code
+        if status_code == 404 and return_if_not_found:
+            return
+        if status_code // 100 == 4:
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Http operation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, ValidationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        try:
+            raise CLIError("Validation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+        except AttributeError:
+            raise CLIError("Validation error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, CloudError):
+        status_code = ex.status_code
+        if status_code == 404 and return_if_not_found:
+            return
+        if status_code // 100 == 4:
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Cloud error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+    raise CLIError("Error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+
+def kubernetes_exception_handler(ex, fault_type, summary, error_message='Error occured while connecting to the kubernetes cluster: ',
+                                 message_for_unauthorized_request='The user does not have required privileges on the kubernetes cluster to deploy Azure Arc enabled Kubernetes agents. Please ensure you have cluster admin privileges on the cluster to onboard.',
+                                 message_for_not_found='The requested kubernetes resource was not found.', raise_error=True):
+    if isinstance(ex, ApiException):
+        status_code = ex.status
+        if status_code // 100 != 2:
+            telemetry.set_user_fault()
```

Comment:
why ApiException is user_fault?

Reply:
These exceptions occur while connecting kubernetes cluster which is owned by the user. Hence if the status code is not 200, it becomes user fault as user does not have sufficient permissions on kubernetes cluster to use connectedk8s cli. The connectedk8s cli requires the user to be admin of the kubernetes cluster it owns